crates/schema/src/type_for_generate.rs

```diff
@@ -161,21 +170,26 @@ impl AlgebraicTypeDef {
         }
     }
 
-    /// Extract all `AlgebraicTypeRef`s that are used in this type into the buffer.
-    fn extract_refs(&self, buf: &mut HashSet<AlgebraicTypeRef>) {
-        match self {
-            AlgebraicTypeDef::Product(ProductTypeDef { elements, .. }) => {
-                for (_, ty) in elements.iter() {
-                    ty.extract_refs(buf);
+    /// Extract all `AlgebraicTypeRef`s that are used in this type into a buffer.
+    /// The buffer may be in arbitrary order, but will not contain duplicates.
+    fn extract_refs(&self) -> SmallVec<[AlgebraicTypeRef; 16]> {
+        EXTRACT_REFS_BUF.with_borrow_mut(|buf| {
+            buf.clear();
```

Comment:
This shouldn't be necessary anymore (not immediately obvious due to the other comment, var being outside the function, but seems to be the case?).